src/NHibernate/Dialect/Dialect.cs

```diff
@@ -1723,6 +1799,24 @@ public virtual string QuoteForSchemaName(string schemaName)
 		}
 
 		/// <summary>
+		/// Quotes a name for being used as a catalogname
+		/// </summary>
+		/// <param name="catalogName">Name of the catalog</param>
+		/// <returns>A Quoted name in the format of OpenQuote + catalogName + CloseQuote</returns>
+		/// <remarks>
+		/// <p>
+		/// If the catalogName is already enclosed in the OpenQuote and CloseQuote then this 
+		/// method will return the catalogName that was passed in without going through any
+		/// Quoting process.  So if catalogName is passed in already Quoted make sure that 
+		/// you have escaped all of the chars according to your DataBase's specifications.
+		/// </p>
+		/// </remarks>
+		public virtual string QuoteForCatalogName(string catalogName)
+		{
+			return IsQuoted(catalogName) ? catalogName : Quote(catalogName);
+		}
```

Comment:
This looks like a general quoting implementation. I definitely will have to check with local sources what is going on with quoting.